server/src/main/scala/controllers/application/exports/WithDesksExport.scala

```diff
@@ -1,44 +1,165 @@
-    val end = start.addDays(1).addMinutes(-1)
-    val summaryForPeriodFn = Exports.queueSummariesFromPortState(airportConfig.nonTransferQueues(terminal(terminalName)), 15, Terminal(terminalName), queryFromPortStateFn(Option(pit.millisSinceEpoch)))
-    Action(exportToCsv(start, end, "desks and queues", terminal(terminalName), None, summaryForPeriodFn))
+          exportStreamingDesksAndQueuesBetweenTimestampsCSV(
+            start,
+            end,
+            terminalName,
+            deploymentsExportStreamForTerminalDates(pointInTime = Option(pit.millisSinceEpoch)),
+            s"desks-and-queues-deps-at-${pit.toISOString()}-for"
+          )
+        case _ =>
+          Action(BadRequest(write(ErrorResponse("Invalid date format"))))
+      }
+    }
+
+  def exportDesksAndQueuesDepsBetweenTimeStampsCSV(startLocalDate: String,
+                                                   endLocalDate: String,
+                                                   terminalName: String): Action[AnyContent] =
+    authByRole(DesksAndQueuesView) {
+      (LocalDate.parse(startLocalDate), LocalDate.parse(endLocalDate)) match {
+        case (Some(startLD), Some(endLD)) =>
+
+          val start = SDate(startLD)
+          val end = SDate(endLD).getLocalNextMidnight.addMinutes(-1)
+          exportStreamingDesksAndQueuesBetweenTimestampsCSV(
+            start,
+            end,
+            terminalName,
+            deploymentsExportStreamForTerminalDates(pointInTime = None),
+            "desks-and-queues-deps"
+          )
+        case _ =>
+          Action(BadRequest(write(ErrorResponse("Invalid date format"))))
+      }
+    }
+
+  def exportStreamingDesksAndQueuesBetweenTimestampsCSV(
+                                                         start: SDateLike,
+                                                         end: SDateLike,
+                                                         terminalName: String,
+                                                         exportSourceFn: (Source[UtcDate, NotUsed], Terminal) =>
+                                                           Source[String, NotUsed],
+                                                         filePrefix: String
+                                                       ): Action[AnyContent] = Action.async {
+
+    val dates = DateRange.utcDateRangeSource(start, end)
+    val terminal = Terminal(terminalName)
+
+    val exportSource: Source[String, NotUsed] = exportSourceFn(dates, terminal)
+    log.info(s"Exporting between $start and $end")
+
+    val fileName = makeFileName(filePrefix, terminal, start, end, airportConfig.portCode)
+
+    Try(sourceToCsvResponse(exportSource, fileName)) match {
+      case Success(value) => Future(value)
+      case Failure(t) =>
+        log.error("Failed to get CSV export", t)
+        Future(BadRequest("Failed to get CSV export"))
+    }
   }
+
+  def deskRecsExportStreamForTerminalDates(pointInTime: Option[MillisSinceEpoch])(
```

Comment:
These two functions are also the same, just calling a different function with the same parameters

Reply:
this is to get round the function they are calling takes implicit parameters, and I couldn't get it two work as a parameter.

Reply:
👍